db/schema_applier.hh

```diff
@@ -129,11 +129,14 @@ public:
     // Makes updates visible. Before calling this function in memory state as observed by other
     // components should not yet change. The function atomically switches current state with
     // new state (the one built in update function).
-    void commit();
+    future<> commit();
     // Notify is called after commit and allows to trigger code which can't provide
     // atomicity either for legacy reasons or causes side effects to an external system
     // (e.g. informing client's driver).
     future<> notify();
+
+private:
+    void commit_on_shard(replica::database& db);
```

Comment:
Should this function be noexcept?

It doesn't seem that the current implementation is guaranteed to never throw (specifically, `database::insert_keyspace` can throw) but I'm not sure the function should be allowed to fail if we want to guarantee atomicity.

Reply:
yes, even flat_hash_map::contains is not marked as noexcept, I don't know how to solve this easily